### PR TITLE
 Network: support peer diagnostics in protocol v2

### LINKED ISSUE #1029: Network: broadcast peer diagnostics in protocol v2
v1 feature we'd like to have in v2 as well.

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes #1029 
Implements https://github.com/nuts-foundation/nuts-specification/pull/184

### CODE REVIEW

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
Looks good, but I suggest adding an integration test as well.

**Comment by reinkrul** [comment]
> Looks good, but I suggest adding an integration test as well.

Integration test found an error: I forgot to register the message hander...

File: network/network_integration_test.go [hunk 1]
```diff
@@ -254,6 +253,40 @@ func TestNetworkIntegration_V2(t *testing.T) {
...
+		// Wait until nodes are connected
+		if !test.WaitFor(t, func() (bool, error) {
+			return len(bootstrap.connectionManager.Peers()) == 1, nil
+		}, defaultTimeout, "time-out while waiting for node 1 and 2 to be connected") {
+			return
+		}
+
+		addTransactionAndWaitForItToArrive(t, "foobar", nutsCrypto.NewTestKey("key"), bootstrap, "integration_node1")
+
+		time.Sleep(100 * time.Millisecond) // wait for diagnostics to be sent
```
Comment (woutslakhorst) [inline network/network_integration_test.go:272]: this could be replaced with a waitFor checking diagnostic length?

**Review by woutslakhorst — APPROVED** [review]
minor improvement possible in test.

Good to see the added test actual caught something broken ;)

**Pull request merged**